lib/monkey/deps/flb_libco/ppc64le.c

```diff
@@ -0,0 +1,253 @@
+  0xc9e30188, /* lfd r15, 392(3) */
+  0xca030190, /* lfd r16, 400(3) */
+  0xca230198, /* lfd r17, 408(3) */
+  0xca4301a0, /* lfd r18, 416(3) */
+  0xca6301a8, /* lfd r19, 424(3) */
+  0xca8301b0, /* lfd r20, 432(3) */
+  0xcaa301b8, /* lfd r21, 440(3) */
+  0xcac301c0, /* lfd r22, 448(3) */
+  0xcae301c8, /* lfd r23, 456(3) */
+  0xcb0301d0, /* lfd r24, 464(3) */
+  0xcb2301d8, /* lfd r25, 472(3) */
+  0xcb4301e0, /* lfd r26, 480(3) */
+  0xcb6301e8, /* lfd r27, 488(3) */
+  0xcb8301f0, /* lfd r28, 496(3) */
+  0xcba301f8, /* lfd r29, 504(3) */
+  0xcbc30200, /* lfd r30, 512(3) */
+  0xcbe30208, /* lfd r31, 520(3) */
+  #endif
+
+  #ifdef __ALTIVEC__
+  0x38a00210, /* li r5, 528        */
+  0x7e842bce, /* stvxl v20, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7ea42bce, /* stvxl v21, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7ec42bce, /* stvxl v22, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7ee42bce, /* stvxl v23, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f042bce, /* stvxl v24, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f242bce, /* stvxl v25, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f442bce, /* stvxl v26, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f642bce, /* stvxl v27, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7f842bce, /* stvxl v28, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7fa42bce, /* stvxl v29, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7fc42bce, /* stvxl v30, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7fe42bce, /* stvxl v31, r4, r5 */
+  0x38a50010, /* addi r5, r5, 16   */
+  0x7ca042a6, /* mfvrsave r5       */
+  0x90a402e0, /* stw r5, 736(4)    */
+  0x38a00210, /* li r5, 528      */
+  0x7e832ace, /* lvxl v20, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7ea32ace, /* lvxl v21, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7ec32ace, /* lvxl v22, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7ee32ace, /* lvxl v23, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f032ace, /* lvxl v24, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f232ace, /* lvxl v25, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f432ace, /* lvxl v26, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f632ace, /* lvxl v27, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7f832ace, /* lvxl v28, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7fa32ace, /* lvxl v29, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7fc32ace, /* lvxl v30, r3, r5 */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x7fe32ace, /* lvxl v31, r3, 5  */
+  0x38a50010, /* addi r5, r5, 16 */
+  0x80a302d0, /* lwz r5, 720(3)  */
+  0x7ca043a6, /* mtvrsave r5     */
+  #endif
+
+  0x4e800020  /* blr             */
+};
+
+/* function call goes directly to code */
+#define CO_SWAP_ASM(x, y) ((void (*)(cothread_t, cothread_t))(uintptr_t)libco_ppc_code)(x, y)
+
+static uint64_t* co_create_(unsigned size, uintptr_t entry) {
```

Comment:
Could you remove this function? It doesn't do anything except for the `malloc` which should be `calloc` to ensure the memory is initialized.